### PR TITLE
DRY bad request response generation.

### PR DESCRIPTION
**Pull request by reitermarkus**
Remove useless redundant checks (i.e. `.is_err()` followed by `.unwrap_err()`) and DRY bad request response generation.

### CODE REVIEW

**Review by tim-hellhake — APPROVED** [review]

File: src/server.rs [hunk 1]
```diff
@@ -560,27 +515,32 @@ fn handle_put_property(
...
-        return HttpResponse::BadRequest().finish();
-    }
+    let arg = if let Some(arg) = args.get(property_name) {
+        arg
+    } else {
+        return HttpResponse::BadRequest().json(bad_request(
+            "Request does not contain property key",
+            Some(json!(args)),
+        ));
+    };
```
Comment (bewee) [inline src/server.rs:533]: I think this is more concise. Unfortunately, I had no chance to try it out. What do you think?

```suggestion
    let arg = args.get(property_name).ok_or_else(|| HttpResponse::BadRequest().json(bad_request(
        "Request does not contain property key",
        Some(json!(args)),
    )))?;
```

**Review by bewee — APPROVED** [review]
Thank you for your contribution! Looks very good :)

File: src/server.rs [hunk 2]
```diff
@@ -560,27 +515,28 @@ fn handle_put_property(
...
+        }
     };
 
-    if !args.contains_key(property_name) {
-        return HttpResponse::BadRequest().finish();
-    }
+    let arg = args.get(property_name).ok_or_else(|| HttpResponse::BadRequest().json(bad_request(
+        "Request does not contain property key",
+        Some(json!(args)),
+    )))?;
```
Comment (tim-hellhake) [inline src/server.rs:529]:
```suggestion
    )));
```

File: src/server.rs [hunk 3]
```diff
@@ -560,27 +515,28 @@ fn handle_put_property(
...
+        }
     };
 
-    if !args.contains_key(property_name) {
-        return HttpResponse::BadRequest().finish();
-    }
+    let arg = args.get(property_name).ok_or_else(|| HttpResponse::BadRequest().json(bad_request(
+        "Request does not contain property key",
+        Some(json!(args)),
+    )));
```
Comment (tim-hellhake) [inline src/server.rs:529]:
```suggestion
    let arg = if let Some(arg) = args.get(property_name) {
        arg
    } else {
        return HttpResponse::BadRequest().json(bad_request(
            "Request does not contain property key",
            Some(json!(args)),
        ));
    };
```

File: src/server.rs (hunk 1 above)
Comment (tim-hellhake) [reply src/server.rs:533]: I agree, but the suggestion doesn't work.
Let's open another PR for this.

**Pull request merged**